src/components/sidekick/index.tsx

```diff
@@ -76,6 +87,45 @@ export class Container extends React.Component<Properties, State> {
     );
   }
 
+  renderTabs(): JSX.Element {
+    return (
+      <div className='sidekick__tabs'>
+        <IconButton
+          className='sidekick__tabs-network'
+          icon={Icons.Network}
+          onClick={this.clickTab.bind(this, Tabs.NETWORK)}
```

Comment:
If we changed our clickTab method above can we get rid of the `bind`?
```
 clickTab = (tab: Tabs) =>
```
```
onClick={this.clickTab(Tabs.NETWORK)}
```

Reply:
if I remove the `.bind` and keep `onClick={this.clickTab(Tabs.NETWORK)}` it will call the method on each render and to solve it I have to do `onClick={() => this.clickTab(Tabs.NETWORK)}` which it will create an anonymous function on each render which do not want.

Reply:
> ...it will create an anonymous function on each render which do not want.

Just as an FYI, `bind` is creating an entirely new on every render too. So, using `bind` isn't really a solution to the problem you've mentioned.

Reply:
you are right, I was wrong about it, it's also written in react documentation.

@dalefukami for learning purposes..., do you have an idea how we can make it better in performance ?

Reply:
There are a couple of strategies that I tend towards:
1. If the list is fairly short and static (like this case) I'll just write 3 separate functions on the class. `this.clickNetworkTab: () => this.clickTab(Tabs.NETWORK)`. They can all call the general case and they're super short so it's pretty obvious that they're just shortcut methods for this purpose.
2. If the list is dynamic then the main strategy is to cache the functions as they're created the first time. I saw this [used in zero-web](https://github.com/m3m3n70/zero-web/blob/development/src/components/gossip-feed/index.tsx#L88) and it's pretty similar to what I've done previously.